.coderabbit.yaml

```diff
@@ -0,0 +1,353 @@
+        structure. Ensure tests avoid complex logic in e2e package and use
+        funcs package for reusable logic. Check that tests don't assume
+        ordering and can run independently. Verify proper cleanup in teardown
+        phases.
+    
+    - path: "**/cluster/**"
+      instructions: |
+        Review Kubernetes manifests for security, resource limits, and proper
+        RBAC. Ensure Helm chart follows best practices. Consider upgrade and
+        rollback scenarios.
+    
+    - path: "**/design/**"
+      instructions: |
+        Focus on architectural decisions, user experience, and long-term
+        maintainability. Ask clarifying questions about design choices and
+        consider alternative approaches. Ensure the design aligns with
+        Crossplane's principles and provides good user experience.
+  
+  # Automatic review settings
+  auto_review:
+    # Enable automatic code reviews on PR creation and updates
+    enabled: true
+    
+    # Enable incremental reviews on each push to the PR
+    auto_incremental_review: true
+    
+    # Skip reviewing draft PRs until they're ready for review
+    drafts: false
+    
+    # Skip reviews if PR title contains these keywords (case-insensitive)
+    ignore_title_keywords:
+      - "wip"
+      - "draft"
+      - "do not merge"
+      - "dnm"
+    
+    # Skip reviews from these automated bot accounts
+    ignore_usernames:
+      - "dependabot[bot]"
+      - "renovate[bot]"
+      - "github-actions[bot]"
+  
+  # Quality gates that run during CodeRabbit's review to check PR readiness
+  pre_merge_checks:
+    # Check PR title for length and descriptiveness
+    title:
+      mode: "warning"
+      requirements: "Keep under 72 characters and be descriptive about what the change does."
+    
+    # Check that PR has a meaningful description
+    description:
+      mode: "warning"
+    
+    # Assess how well the PR addresses any linked GitHub issues
+    issue_assessment:
+      mode: "warning"
+    
+    # Check that exported functions and types have proper documentation
+    docstrings:
+      mode: "warning"
+      threshold: 80
+    
+    # Custom checks specific to Crossplane development practices
+    custom_checks:
+      - name: "Breaking Changes"
+        mode: "warning"
+        instructions: "Check if this change introduces any breaking changes to APIs, CLI commands, or behavior. If so, ensure proper deprecation notices and migration guides are provided."
+      
+      - name: "Feature Gate Consideration"
+        mode: "warning"
+        instructions: "For new features, consider if they should be behind a feature gate, especially if they're experimental or could impact stability."
+      
+      - name: "Documentation Updates"
+        mode: "warning"
+        instructions: "Ensure that user-facing changes include appropriate documentation updates or clearly indicate what documentation needs to be updated."
```

Comment:
will it be able to do this effectively, since docs are in a different repo?

Reply:
Probably not - I actually thought I removed this. Will do.